src/test/java/com/nimiq/NimiqClientTest.java

```diff
@@ -0,0 +1,467 @@
+
+package com.nimiq;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * NimiqClientTest<br/>
+ * Assumes a Nimiq node is started locally:
+ *
+ * <pre>
+ * nodejs index.js --protocol=dumb --type=full --network=test --rpc
+ * </pre>
+ *
+ * requres a test account charged with some NIM
+ */
+public class NimiqClientTest {
+
+    private static final String RPC_URL = "http://localhost:8648/";
+
+    private static final String GENESIS_BLOCK = "0001000000000000000000000000000000000000000000"
+            + "00000000000000000000000000000000000000000000000000000000000000000000000000000000"
+            + "000000f6ba2bbf7e1478a209057000471d73fbdc28df0b717747d929cfde829c4120f62e02da3d16"
+            + "2e20fa982029dbde9cc20f6b431ab05df1764f34af4c62a4f2b33f1f010000000000015ac3185f00"
+            + "0134990001000000000000000000000000000000000000000007546573744e657400000000";
+    private static final String BURN_ADDRESS = "NQ07 0000 0000 0000 0000 0000 0000 0000 0000";
+    private static final String FAUCET_ADDRESS = "NQ02 YP68 BA76 0KR3 QY9C SF0K LP8Q THB6 LTKU";
+    private static final String VESTING_ACCOUNT_ADDRESS = "NQ74 919Y 3HK2 XFKP MM7C YF2S EUG3 PTV2 S2PF";
+    private static final String HTLC_ADDRESS = "NQ26 4XVR N8BB THR4 RB81 MF3V C5DM 49V5 7TQU";
+    private static final int BLOCK_NUMBER = 36360;
+    private static final String BLOCK_HASH = "2c9d91f8d7061f2386e4221370ed9dec53b96c9d42026f20bd8853ed74f2b2ef";
+    private static final String TX1_HASH = "69834d7b39e05a3b077124dfebde54d0ced636bc98c46ea68fbf885e34a0fbeb";
+    private static final String TX2_HASH = "eb4b690aa5daf0c416f5cd7a7b8f9d4100a865b858a23de7d99b9dcfe58086cd";
+    private static final String UNKNOWN_HASH = "0000000000000000000000000000000000000000000000000000000000000000";
+    private static final int TX_SIZE = 138;
+    private static final long MIN_FEE = TX_SIZE; // 1 luna/byte
+
+    private static NimiqClient client;
+
+    @BeforeClass
+    public static void createClient() throws MalformedURLException {
+        final URL url = new URL(RPC_URL);
+        client = new NimiqClientFactory(url).getClient();
+    }
+
+    // Network
+
+    @Test
+    public void testPeerCount() {
+        assertTrue(client.peerCount() > 0);
+    }
+
+    @Test
+    public void testSyncing() {
+        final SyncStatus status = client.syncing();
```

Comment:
I agree that syncing() is not a standard Java name. On the other hand, to be consistent we need to rename other methods as well: consensus(), peerCount(), blockNumber(). This is not an issue at all, I just kept the original names. If we change there will be some names that match the original RPC methods and some that don't (because not all original names are verbs).

Is there the same issue in other clients (like Go) ?

Reply:
Very good point. I'll check. There's two sides to it: on one hand you want to keep the methods in sync with the core API - but on the other you want it to be in-line with the coding standards of this lang. I tend towards the second: follow the standards of this language. You're a Java coder, wdyt / prefer?

Reply:
Ideally, I'd like all RPC methods to be verbs like in [Bitcoin](https://en.bitcoin.it/wiki/Original_Bitcoin_client/API_calls_list). But it's too late to change the protocol, isn't it ? :)

Reply:
Renamed in https://github.com/nimiq-community/java-client/pull/1/commits/6db8635fef6212b872ae31bbed9d3d965a1761f8